### ISSUE TITLE
Bash syntax highlighting bug when using comments in backticks

### ISSUE THREAD

**Issue by ssynix** [Microsoft/vscode#34643]
- VSCode Version: 1.16.1
- OS Version: MacOS Sierra

**Steps to Reproduce:**

I use the following style to insert inline documentation for long bash commands and their cryptic args.

```bash
foo                             \
  -d `# we use d for something` \
  -c `# we use c for something` \
  ...
```

The same snippet on vscode breaks syntax highlighting in other parts of the code, because it doesn't correctly detect the comment statement inside backticks. Use keywords (`for`, `in`), or quotes inside the comment statements to see the bug.

**Comment by aeschli** [comment]
This issue was moved to textmate/shellscript.tmbundle#44

**Comment by dsifford** [comment]
@aeschli Related question: Is there a reason for not using Atom's tmLangugage implementation in shellscript like VSCode is for a majority of the other languages? The Atom repo looks like it's much more active than the textmate one.

(Also, on a personal note: It's a much more pleasant experience to contribute fixes to Atom languages because they use CSON, rather than XML).

Thanks in advance for the clarification. 😄

**Comment by aeschli** [comment]
@ssynix We can switch to Atom if people like that grammar better. There are only few open bugs on textmate/shellscript.tmbundle#44 so from our side, there's no urgency.

**Comment by dsifford** [comment]
@aeschli If we were to move over to the Atom syntax, I'd personally fix whatever issues are outstanding in shellscript including [these few that I ran into](https://github.com/dracula/visual-studio-code/blob/master/known_issues.md#shell). I spent the last few days going through some of the bugs I've found and [submitting](https://github.com/atom/language-sass/pull/234) [PRs](https://github.com/atom/language-less/pull/82) [to](https://github.com/atom/language-css/pull/127) [Atom](https://github.com/atom/language-php/pull/285) for their grammars and I've already established rapport with them so the process is just a bit easier for me.

Similarly to this, I've been blocked from being able to help fix `Make`, `Ruby`, `Perl`, and `Powershell` (the latter being only because the ergonomics is difficult being that I'd need to compile a development version of VSCode _and_ deal with XML to contribute fixes).

Just a suggestion. Thanks for considering.

**Comment by aeschli** [comment]
@dsifford I switch the shell grammar to use https://github.com/atom/language-shellscript.

This issue seems to be fixed by that.

**Comment by dsifford** [comment]
@ssynix Awesome, I appreciate the switch!

Question for you now that I've got your attention: What is the current process (and frequency) that VSCode pulls grammars from atom? I've noticed on a number of occasions recently that there will be issues in VSCode, but not in atom (presumptively because VSCode's grammars are behind Atom's)

I assume (for the stable channel) the languages are synced the day of release. But what about for insiders? Is this a daily compile task? 

Thanks for the clarification 😊
